modules/core/src/lib/attribute.js

```diff
@@ -170,6 +170,11 @@ export default class Attribute extends BaseAttribute {
 
       this.constant = false;
       this.value = new ArrayType(this.size * allocCount);
+
+      if (this.buffer && this.buffer.byteLength < this.value.byteLength) {
```

Comment:
@ibgreen @tsherif This is the part I'm hoping to get some ideas - in the incremental update use case (e.g. data loaded by chunks), every time we reallocate the buffer, we need to re-upload the entire value array.

Reply:
Is it possible to know ahead of time in that incremental case what the total size of the data will be?

Reply:
Generally, no.

Alternatively, every time overflow happens, we could over-allocate the buffer instead of setting it to the exact new size.

Reply:
That makes sense to me. Something like doubling the size of the buffer on each overflow.

Reply:
How sensitive should we be regarding GPU memory usage?

Reply:
It could be a problem on mobile, but I think we have some leeway because we don't use textures much. It would be helpful to profile some typical use cases and see what our current memory usage is like.